index.js

```diff
@@ -142,22 +160,16 @@ SignalhubWs.prototype.close = function (cb) {
       socket.close()
     })
   })
-}
 
-SignalhubWs.prototype._closeChannels = function (cb) {
-  if (this.closed) {
-    if (cb) process.nextTick(cb)
-    return
-  }
-  this.closed = true
-
-  if (cb) {
-    this.on('close', cb)
-  }
+  this.once('close:channels', () => {
+    this.emit('close')
+  })
+}
 
+SignalhubWs.prototype._closeChannels = function () {
   const len = this.channels.size
   if (len === 0) {
-    this.emit('close')
+    this.emit('close:channels')
```

Comment:
why not `close` directly? It's forwarding the event to close?

Reply:
We can close directly, I'll fix it.